client/state/data-layer/wpcom-http/index.js

```diff
@@ -45,18 +54,32 @@ const queueRequest = ( { dispatch }, action, next ) => {
 		{ path, formData },
 		query,
 		method === 'POST' && body,
-		( error, data ) => !! error
-			? onFailure && dispatch( extendAction( onFailure, failureMeta( error ) ) )
-			: onSuccess && dispatch( extendAction( onSuccess, successMeta( data ) ) )
+		( error, data ) => {
+			const {
+				failures,
+				nextData,
+				nextError,
+				shouldAbort,
+				successes
+			} = processEgress( action, { dispatch }, data, error );
+
+			if ( true === shouldAbort ) {
```

Comment:
I didn't have any specifics in mind when I created this. it seemed like an obvious piece of the API to add and it doesn't introduce much complexity.

use cases could involve testing applications which randomly drop responses to see how the app performs, situations like you bring up with data-freshness, etc…